lib/improver/tests/cube_combiner/test_CubeCombiner.py

```diff
@@ -0,0 +1,335 @@
+        self.assertIsInstance(result[1], Cube)
+
+
+class Test_amend_metadata(IrisTest):
+
+    """Test the amend_metadata method."""
+
+    def test_basic(self):
+        """Test that the function returns a Cube. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        result = plugin.amend_metadata(cube, 'new_cube_name', np.dtype,
+                                       None, None)
+        self.assertIsInstance(result, Cube)
+        self.assertEqual(result.name(), 'new_cube_name')
+
+    def test_attributes_updated(self):
+        """Test amend_metadata  updates attributes OK. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        attributes = {'relative_to_threshold': 'between'}
+        result = plugin.amend_metadata(cube, 'new_cube_name', np.dtype,
+                                       None, attributes)
+        self.assertEqual(result.attributes['relative_to_threshold'],
+                         'between')
+
+    def test_attributes_added(self):
+        """Test amend_metadata  updates attributes OK. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        attributes = {'new_attribute': 'new_value'}
+        result = plugin.amend_metadata(cube, 'new_cube_name', np.dtype,
+                                       None, attributes)
+        self.assertEqual(result.attributes['new_attribute'],
+                         'new_value')
+
+    def test_attributes_deleted(self):
+        """Test amend_metadata  updates attributes OK. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        attributes = {'relative_to_threshold': 'delete'}
+        result = plugin.amend_metadata(cube, 'new_cube_name', np.dtype,
+                                       None, attributes)
+        self.assertEqual('relative_to_threshold' in result.attributes,
+                         False)
+
+    def test_coords_updated(self):
+        """Test amend_metadata updates coords correctly. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        coords = {'threshold': {'points': [2.0], 'bounds': [0.1, 2.0]}}
+        result = plugin.amend_metadata(cube, 'new_cube_name', np.dtype,
+                                       coords, None)
+        self.assertArrayEqual(result.coord('threshold').points,
+                              np.array([2.0]))
+
+    def test_coords_deleted(self):
+        """Test amend_metadata updates coords correctly. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        coords = {'threshold': 'delete'}
+        result = plugin.amend_metadata(cube, 'new_cube_name', np.dtype,
+                                       coords, None)
+        found_key = 'threshold' in [coord.name() for coord in result.coords()]
+        self.assertArrayEqual(found_key,
+                              False)
+
+    def test_coords_deleted_fails(self):
```

Comment:
I think that there are three possible ways that a ValueError can be raised within the amend_metadata method, which each need unit tests.